Userland/Libraries/LibWeb/HTML/Navigable.cpp

```diff
@@ -1574,6 +1574,10 @@ WebIDL::ExceptionOr<void> Navigable::navigate_to_a_fragment(URL::URL const& url,
 // https://html.spec.whatwg.org/multipage/browsing-the-web.html#evaluate-a-javascript:-url
 WebIDL::ExceptionOr<JS::GCPtr<DOM::Document>> Navigable::evaluate_javascript_url(URL::URL const& url, URL::Origin const& new_document_origin, String navigation_id)
 {
+    // AD-HOC: Don't run if scripting is disabled.
+    if (active_document()->is_scripting_disabled())
+        return nullptr;
```

Comment:
This doesn't seem like the right fix. We already check if scripting is enabled below when we call `script->run()`. The crash is because that method returns an empty value:
```c++
// 2. Check if we can run script with settings. If this returns "do not run" then return NormalCompletion(empty).
if (settings.can_run_script() == RunScriptDecision::DoNotRun)
    return JS::normal_completion({});
```

And we then we blindly dereference that empty `Optional<Value>` when we do `evaluation_status.value()->is_string()`.

So I think the appropriate fix is to just check if that value is empty before dereferencing it:
```c++
// 9. If evaluationStatus is a normal completion, and evaluationStatus.[[Value]] is a String, then set result to evaluationStatus.[[Value]].
if (evaluation_status.type() == JS::Completion::Type::Normal && evaluation_status.value().has_value() && evaluation_status.value()->is_string()) {
    result = evaluation_status.value()->as_string().utf8_string();
} else {
    // 10. Otherwise, return null.
    return nullptr;
}
```

Reply:
I see, thanks. Updated and verified the crash doesn't occur.